### PR TITLE
Reduce the level of structural statements to make it look more comfortable. - Composer\XdebugHandsler:check

### PR DESCRIPTION
**Pull request by medz**

### CODE REVIEW

File: src/Composer/XdebugHandler.php [hunk 1]
```diff
@@ -63,12 +63,11 @@ public function check()
...
         if ($this->needsRestart($args[0])) {
-            if ($this->prepareRestart()) {
-                $command = $this->getCommand();
-                $this->restart($command);
+            if (! $this->prepareRestart()) {
+                return ;
             }
 
-            return;
+            $this->restart($this->getCommand());
```
Comment (stof) [inline src/Composer/XdebugHandler.php]: this does not have the return anymore at this end of this `if ($this->needsRestart($args[0])) {`, so it is changing the code

**Comment by stof** [comment]
I don't see how this reduces the complexity of the method. The cyclomatic complexity is exactly the same, as you still have the same number of conditions (replacing a `if/else` by a ternary operator does not reduce complexity at all)

**Comment by medz** [comment]
@stof I am phrased `phpmd` principle of correction, the visual will be better, their logic is the same.

**Comment by medz** [comment]
@stof And the conditions are not established when the direct return than if the implementation of the conditions set up after the operation seems more beautiful.

**Comment by Seldaek** [comment]
Not so sure whether this is an improvement, but anyway this code is being migrated to the composer/xdebug-handler repo so if you still want to do this please send the PR again there.

**Pull request closed without merging**